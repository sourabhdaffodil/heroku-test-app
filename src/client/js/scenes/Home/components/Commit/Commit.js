/* eslint no-underscore-dangle: 0 */
import React, { Component } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { media } from '../../../../theme/style-utils';
import Button from '../../../../components/Button/Button';
import PillLeft from '../../../../components/Button/PillLeft/PillLeft';
import PillRight from '../../../../components/Button/PillRight/PillRight';
import Icon from '../../../../components/Icon/Icon';
import Link from '../../../../components/Link/Link';
import { monthNames } from '../../services/months';

const Li = styled.li`
  display: flex;
  list-style: none;
  padding: 15px;

  &:not(:last-child) {
    border-bottom: 1px solid ${props => props.theme.color('background')};
  }

  &:hover {
    background-color: ${props => props.theme.color('primary', 'lighter')};
  }

  i {
    color: ${props => props.theme.color('primary')};
  }
`;

const HashBtnContainer = styled.div`
  ${media.phone`
    display: none;
  `}
`;

const Body = styled.div`
  flex-grow: 1;
`;

const Message = styled.div`
  display: flex;
`;

const Img = styled.img`
  width: 36px;
  height: 36px;
  border-radius: 3px;
  margin-right: 10px;
  
  ${media.phone`
    width: 20px;
    height: 20px;
  `}
`;

const Title = styled.div`
  font-weight: ${props => props.theme.weight('semiBold')};
  ${props => props.theme.textSetting('nr')};
  margin-bottom: 5px;
  display: flex;
  align-items: center;
  
  ${media.phone`
    font-size: ${props => props.theme.textSetting('sm').fontSize};
    margin-bottom: 0;
  `}
`;

const ExpandDescriptionBtn = styled.button`
  padding: 0 10px 5px;
  background-color: ${props => props.theme.color('background')};
  font-size: ${props => props.theme.textSetting('xs').fontSize};
  margin-left: 10px;
  border: 0;
  cursor: pointer;
  
  ${media.phone`
    
  `}

  &:active {
    background-color: ${props => props.theme.color('primary', 'light')};
    color: ${props => props.theme.contrast('dark')};
  }
`;

const Author = styled.span`
  ${props => props.theme.textSetting('sm')};
  
  ${media.phone`
    color: ${props => props.theme.contrast('light', 'secondary')};
    ${props => props.theme.textSetting('xs')};
  `}
`;

const AuthorName = styled.span`
  font-weight: ${props => props.theme.weight('semiBold')}
`;

const HashAlias = styled.span`
  color: ${props => props.theme.color()};
`;

const CommitedBy = styled.span`
  display: none;
  
  ${media.phone`
    display: block;
  `}
`;

const CommitedOn = styled.span`
  ${media.phone`
    display: none;
  `}
`;

const Description = styled.p`
  ${props => props.theme.textSetting('sm')}
  
  ${media.phone`
    ${props => props.theme.textSetting('xs')}
  `}
`;

class Commit extends Component {
  constructor(props) {
    super(props);

    this.state = {
      expanded: false,
    };
  }

  getCommitHash() {
    const { commit } = this.props;
    return commit._id.substring(commit._id.length - 7);
  }

  toggleDescription() {
    this.setState({
      expanded: !this.state.expanded,
    });
  }

  renderDescription() {
    if (this.state.expanded) {
      return (
        <Description>{this.props.commit.description}</Description>
      );
    }

    return null;
  }

  renderExpandButton() {
    const { commit } = this.props;
    if (commit.description && commit.description.length > 0) {
      return (
        <ExpandDescriptionBtn onClick={() => this.toggleDescription()}>
          ...
        </ExpandDescriptionBtn>
      );
    }

    return null;
  }

  renderHashBtn() {
    const { commit } = this.props;
    if (commit.link && commit.link.length > 0) {
      return (
        <HashBtnContainer>
          <Link url={commit.link}>
            <Button light small title="Related link">
              <PillLeft>
                <Icon icon="link" />
              </PillLeft>
              <PillRight>
                <HashAlias>
                  {this.getCommitHash()}
                </HashAlias>
              </PillRight>
            </Button>
          </Link>
        </HashBtnContainer>
      );
    }

    return null;
  }

  renderHashLink() {
    const { commit } = this.props;
    if (commit.link && commit.link.length > 0) {
      return (
        <span>
          &nbsp;-
          &nbsp;<Link url={commit.link} className="link">{this.getCommitHash()}</Link>
        </span>
      );
    }

    return null;
  }

  render() {
    const { commit } = this.props;
    if (Object.keys(commit).length) {
      return (
        <Li>
          <Img src={commit.image} width="50px" height="50px" alt={commit.title} />
          <Body>
            <Title>
              <span>{commit.title}</span>
              {this.renderExpandButton()}
            </Title>
            <Message>
              <Author>
                <CommitedBy>
                  Commited by
                  &nbsp;<AuthorName>{commit.author}</AuthorName>
                  {this.renderHashLink()}
                </CommitedBy>
                <CommitedOn>
                  <AuthorName>{commit.author}</AuthorName>
                  &nbsp;commited on {monthNames[commit.date.getMonth()]}
                  &nbsp;{commit.date.getFullYear()}
                </CommitedOn>
              </Author>
            </Message>
            {this.renderDescription()}
          </Body>
          {this.renderHashBtn()}
        </Li>
      );
    }

    return null;
  }
}

Commit.propTypes = {
  commit: PropTypes.object.isRequired,
};

export default Commit;
