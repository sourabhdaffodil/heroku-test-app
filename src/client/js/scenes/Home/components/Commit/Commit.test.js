import React from 'react';
import { shallow } from 'enzyme';
import Commit from './Commit';

describe('Commit component', () => {
  it('Must render an empty list', () => {
    const wrapper = shallow(<Commit commit={{}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Must render the details of a commits', () => {
    const commit = {
      _id: '59d678d9f51fea02ea1209b7',
      date: new Date('2016-06-04'),
      title: 'A title',
      description: 'Description...',
      image: 'https://avatars0.githubusercontent.com/u/4216535?v=4&s=72',
      link: 'sample.com',
    };
    const wrapper = shallow(<Commit commit={commit} />);
    expect(wrapper).toMatchSnapshot();
  });
});
