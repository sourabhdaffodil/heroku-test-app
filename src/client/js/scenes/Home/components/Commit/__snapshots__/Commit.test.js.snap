// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commit component Must render an empty list 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Commit
    commit={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Commit component Must render the details of a commits 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Commit
    commit={
      Object {
        "_id": "59d678d9f51fea02ea1209b7",
        "date": 2016-06-04T00:00:00.000Z,
        "description": "Description...",
        "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
        "link": "sample.com",
        "title": "A title",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.img
          alt="A title"
          height="50px"
          src="https://avatars0.githubusercontent.com/u/4216535?v=4&s=72"
          width="50px"
        />,
        <styled.div>
          <styled.div>
            <span>
              A title
            </span>
            <styled.button
              onClick={[Function]}
            >
              ...
            </styled.button>
          </styled.div>
          <styled.div>
            <styled.span>
              <styled.span>
                Commited by  
                <styled.span />
                <span>
                   -  
                  <Link
                    className="link"
                    url="sample.com"
                  >
                    a1209b7
                  </Link>
                </span>
              </styled.span>
              <styled.span>
                <styled.span />
                 commited on 
                June
                 
                2016
              </styled.span>
            </styled.span>
          </styled.div>
        </styled.div>,
        <styled.div>
          <Link
            className=""
            url="sample.com"
          >
            <Button
              light={true}
              small={true}
              title="Related link"
            >
              <PillLeft>
                <Icon
                  icon="link"
                />
              </PillLeft>
              <PillRight>
                <styled.span>
                  a1209b7
                </styled.span>
              </PillRight>
            </Button>
          </Link>
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "alt": "A title",
          "height": "50px",
          "src": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
          "width": "50px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              <span>
                A title
              </span>
              <styled.button
                onClick={[Function]}
              >
                ...
              </styled.button>
            </styled.div>,
            <styled.div>
              <styled.span>
                <styled.span>
                  Commited by  
                  <styled.span />
                  <span>
                     -  
                    <Link
                      className="link"
                      url="sample.com"
                    >
                      a1209b7
                    </Link>
                  </span>
                </styled.span>
                <styled.span>
                  <styled.span />
                   commited on 
                  June
                   
                  2016
                </styled.span>
              </styled.span>
            </styled.div>,
            null,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <span>
                  A title
                </span>,
                <styled.button
                  onClick={[Function]}
                >
                  ...
                </styled.button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "A title",
                },
                "ref": null,
                "rendered": "A title",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "...",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "...",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": <styled.span>
                <styled.span>
                  Commited by  
                  <styled.span />
                  <span>
                     -  
                    <Link
                      className="link"
                      url="sample.com"
                    >
                      a1209b7
                    </Link>
                  </span>
                </styled.span>
                <styled.span>
                  <styled.span />
                   commited on 
                  June
                   
                  2016
                </styled.span>
              </styled.span>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.span>
                    Commited by  
                    <styled.span />
                    <span>
                       -  
                      <Link
                        className="link"
                        url="sample.com"
                      >
                        a1209b7
                      </Link>
                    </span>
                  </styled.span>,
                  <styled.span>
                    <styled.span />
                     commited on 
                    June
                     
                    2016
                  </styled.span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "Commited by  ",
                      <styled.span />,
                      <span>
                         -  
                        <Link
                          className="link"
                          url="sample.com"
                        >
                          a1209b7
                        </Link>
                      </span>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Commited by  ",
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "class",
                      "props": Object {
                        "children": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          " -  ",
                          <Link
                            className="link"
                            url="sample.com"
                          >
                            a1209b7
                          </Link>,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        " -  ",
                        Object {
                          "instance": null,
                          "key": null,
                          "nodeType": "function",
                          "props": Object {
                            "children": "a1209b7",
                            "className": "link",
                            "url": "sample.com",
                          },
                          "ref": null,
                          "rendered": "a1209b7",
                          "type": [Function],
                        },
                      ],
                      "type": "span",
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <styled.span />,
                      " commited on ",
                      "June",
                      " ",
                      2016,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "class",
                      "props": Object {
                        "children": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    " commited on ",
                    "June",
                    " ",
                    2016,
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          null,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <Link
            className=""
            url="sample.com"
          >
            <Button
              light={true}
              small={true}
              title="Related link"
            >
              <PillLeft>
                <Icon
                  icon="link"
                />
              </PillLeft>
              <PillRight>
                <styled.span>
                  a1209b7
                </styled.span>
              </PillRight>
            </Button>
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <Button
              light={true}
              small={true}
              title="Related link"
            >
              <PillLeft>
                <Icon
                  icon="link"
                />
              </PillLeft>
              <PillRight>
                <styled.span>
                  a1209b7
                </styled.span>
              </PillRight>
            </Button>,
            "className": "",
            "url": "sample.com",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <PillLeft>
                  <Icon
                    icon="link"
                  />
                </PillLeft>,
                <PillRight>
                  <styled.span>
                    a1209b7
                  </styled.span>
                </PillRight>,
              ],
              "light": true,
              "small": true,
              "title": "Related link",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "children": <Icon
                    icon="link"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "icon": "link",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "children": <styled.span>
                    a1209b7
                  </styled.span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "a1209b7",
                  },
                  "ref": null,
                  "rendered": "a1209b7",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.img
            alt="A title"
            height="50px"
            src="https://avatars0.githubusercontent.com/u/4216535?v=4&s=72"
            width="50px"
          />,
          <styled.div>
            <styled.div>
              <span>
                A title
              </span>
              <styled.button
                onClick={[Function]}
              >
                ...
              </styled.button>
            </styled.div>
            <styled.div>
              <styled.span>
                <styled.span>
                  Commited by  
                  <styled.span />
                  <span>
                     -  
                    <Link
                      className="link"
                      url="sample.com"
                    >
                      a1209b7
                    </Link>
                  </span>
                </styled.span>
                <styled.span>
                  <styled.span />
                   commited on 
                  June
                   
                  2016
                </styled.span>
              </styled.span>
            </styled.div>
          </styled.div>,
          <styled.div>
            <Link
              className=""
              url="sample.com"
            >
              <Button
                light={true}
                small={true}
                title="Related link"
              >
                <PillLeft>
                  <Icon
                    icon="link"
                  />
                </PillLeft>
                <PillRight>
                  <styled.span>
                    a1209b7
                  </styled.span>
                </PillRight>
              </Button>
            </Link>
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "alt": "A title",
            "height": "50px",
            "src": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
            "width": "50px",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <span>
                  A title
                </span>
                <styled.button
                  onClick={[Function]}
                >
                  ...
                </styled.button>
              </styled.div>,
              <styled.div>
                <styled.span>
                  <styled.span>
                    Commited by  
                    <styled.span />
                    <span>
                       -  
                      <Link
                        className="link"
                        url="sample.com"
                      >
                        a1209b7
                      </Link>
                    </span>
                  </styled.span>
                  <styled.span>
                    <styled.span />
                     commited on 
                    June
                     
                    2016
                  </styled.span>
                </styled.span>
              </styled.div>,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <span>
                    A title
                  </span>,
                  <styled.button
                    onClick={[Function]}
                  >
                    ...
                  </styled.button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "A title",
                  },
                  "ref": null,
                  "rendered": "A title",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "...",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "...",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": <styled.span>
                  <styled.span>
                    Commited by  
                    <styled.span />
                    <span>
                       -  
                      <Link
                        className="link"
                        url="sample.com"
                      >
                        a1209b7
                      </Link>
                    </span>
                  </styled.span>
                  <styled.span>
                    <styled.span />
                     commited on 
                    June
                     
                    2016
                  </styled.span>
                </styled.span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.span>
                      Commited by  
                      <styled.span />
                      <span>
                         -  
                        <Link
                          className="link"
                          url="sample.com"
                        >
                          a1209b7
                        </Link>
                      </span>
                    </styled.span>,
                    <styled.span>
                      <styled.span />
                       commited on 
                      June
                       
                      2016
                    </styled.span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        "Commited by  ",
                        <styled.span />,
                        <span>
                           -  
                          <Link
                            className="link"
                            url="sample.com"
                          >
                            a1209b7
                          </Link>
                        </span>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Commited by  ",
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "class",
                        "props": Object {
                          "children": undefined,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            " -  ",
                            <Link
                              className="link"
                              url="sample.com"
                            >
                              a1209b7
                            </Link>,
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          " -  ",
                          Object {
                            "instance": null,
                            "key": null,
                            "nodeType": "function",
                            "props": Object {
                              "children": "a1209b7",
                              "className": "link",
                              "url": "sample.com",
                            },
                            "ref": null,
                            "rendered": "a1209b7",
                            "type": [Function],
                          },
                        ],
                        "type": "span",
                      },
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        <styled.span />,
                        " commited on ",
                        "June",
                        " ",
                        2016,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": null,
                        "nodeType": "class",
                        "props": Object {
                          "children": undefined,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      " commited on ",
                      "June",
                      " ",
                      2016,
                    ],
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            null,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <Link
              className=""
              url="sample.com"
            >
              <Button
                light={true}
                small={true}
                title="Related link"
              >
                <PillLeft>
                  <Icon
                    icon="link"
                  />
                </PillLeft>
                <PillRight>
                  <styled.span>
                    a1209b7
                  </styled.span>
                </PillRight>
              </Button>
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": <Button
                light={true}
                small={true}
                title="Related link"
              >
                <PillLeft>
                  <Icon
                    icon="link"
                  />
                </PillLeft>
                <PillRight>
                  <styled.span>
                    a1209b7
                  </styled.span>
                </PillRight>
              </Button>,
              "className": "",
              "url": "sample.com",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <PillLeft>
                    <Icon
                      icon="link"
                    />
                  </PillLeft>,
                  <PillRight>
                    <styled.span>
                      a1209b7
                    </styled.span>
                  </PillRight>,
                ],
                "light": true,
                "small": true,
                "title": "Related link",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "children": <Icon
                      icon="link"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "icon": "link",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "children": <styled.span>
                      a1209b7
                    </styled.span>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": "a1209b7",
                    },
                    "ref": null,
                    "rendered": "a1209b7",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
